### ISSUE TITLE
[Bug] Bar chart item overlap grid

### ISSUE THREAD

**Issue by ygs5933798** [apache/echarts#18793]
### Version

latest

### Link to Minimal Reproduction

https://codepen.io/Sam-Yang-the-selector/pen/MWzbmYE

### Steps to Reproduce

My case is bar chart with xAxis type is value, and the min value of xAxis is a value, option like image below

![image](https://github.com/apache/echarts/assets/127940020/c8445849-0ed5-4d8d-aee8-0ae3119699fe)

### Current Behavior

chart item overlap

### Expected Behavior

the first bar item need to display half width of it, because left part is overflow grid.

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

**Comment by Ovilia** [comment]
This is the expected result since you set min: 110.

**Comment by ygs5933798** [comment]
> This is the expected result since you set min: 110.

Thanks for reply

Yes, the min data is 110, but do we have any way to hide the part of bar which overlap grid?

**Comment by michael-s-molina** [comment]
@Ovilia I agree with @ygs5933798 that this does not look good. We had [user reports](https://github.com/apache/superset/issues/26254) in Superset about the same problem. @ygs5933798 manually set 110 but this also happens when using `dataMin` and `dataMax` which are automatically calculated. Can we try to come up with an improved display? Maybe moving the ticks at both ends to accommodate the bars?

**Comment by michael-s-molina** [comment]
@Ovilia Look how this is handled differently when comparing `category` and `value` even though both set `min` and `max`.

Type `category` with `min` and `max` defined:

<img width="1782" alt="Screenshot 2023-12-13 at 11 41 03" src="https://github.com/apache/echarts/assets/70410625/7da12001-0289-4a47-8204-6ec03550aaa7">

Type `value` with `min` and `max` defined:

<img width="1783" alt="Screenshot 2023-12-13 at 11 43 31" src="https://github.com/apache/echarts/assets/70410625/6ec81fd4-a3c1-44f6-a614-c2800022d061">

**Comment by michael-s-molina** [comment]
I found out that you can use [scale](https://echarts.apache.org/en/option.html#xAxis.scale) to achieve this [behavior](https://echarts.apache.org/examples/en/editor.html?c=bar-simple&code=PYBwLglsB2AEC8sDeAoWsAeBBDEDOAXMmurGAJ4gCmRA5AG4CGANgK5W0A0J6eAxixpkATu27oAvuNjkc-IqlJlKQhi3ZcSUknirCIVQrADaPYkvQATRmEZFjARgBMABk6xXb2A4CsXgBxeAOxezqEAzC4AutJKFNR0AEaMwppKEiRRKBIA3EA).

**Comment by michael-s-molina** [comment]
@Ovilia @villebro Can we prioritize a fix for this problem in ECharts 6.0? It is quite apparent in Superset.

**Comment by villebro** [comment]
> Can we prioritize a fix for this problem in ECharts 6.0? It is easily reproducible in Superset [apache/superset#26254](https://github.com/apache/superset/issues/26254)

@michael-s-molina would you be able to convey this to the [Dev list discussion](https://lists.apache.org/thread/zbh71t9rz8n4fyryds9xgbb44rr215pv)? I will see if I can contribute to 6.0 efforts, and this could be a good one to start with, as it seems to not only affect Superset, but the community in general.
